common/http/src/retry.rs

```diff
@@ -99,39 +99,51 @@ impl Retry {
     }
 
     async fn fill_waiting(&self) -> Result<(), Error> {
-        let mut files = read_dir(&self.directory).await?;
-        while let Some(file) = files.next_entry().await? {
-            let path = file.path();
-            if path.is_dir() {
-                continue;
+        match read_dir(&self.directory).await {
+            Err(e) => {
+                // bad retry_dir is fatal error
+                error!("Error: {:?}", e);
+                error!(
+                    "Invalid Retry Directory: '{:?}'. Check configuration!",
+                    self.directory
+                );
+                std::process::exit(e.raw_os_error().unwrap_or(255));
```

Comment:
Yes, if it is not fatal error. In this case it is fatal error. The app cannot continue. Cannot rely on graceful shutdown.

Reply:
It's fatal for the retry stream yes, but I don't think that blocks the rest of the program from being able to shutdown gracefully.

Not shutting them down gracefully could, for example, cause rocksdb to be left in an inconsistent state.

Reply:
We can use `panic!` instead to exit in a fatal error, but IMO I think this particular case can be handled gracefully.

Reply:
This very abnormal rare case - user deleted retry folder. 
Db is using WAL log. We do not want to finish any transaction or continue spinning in agent as it will cause log loss.

Reply:
https://stackoverflow.com/questions/39228685/when-is-stdprocessexit-o-k-to-use

From this, `panic!` appears to do more than just print the stack trace. We probably do need to do more than just that to properly exit the entire program in a fatal error, but from what I'm reading we should not be using `std::process::exit` in the way we're using it.

Reply:
Again, this is not normal OK exit case. Any way you try delay exit will cause log loss. 
Agent doe snot have any persistent state except db. We do not want to finish anything in this case.